src/rewards/voters.rs

```diff
@@ -30,17 +42,241 @@ pub fn calculate_stake<'address>(
     (total_stake, stake_per_voter)
 }
 
-pub fn calculate_reward_share<'address>(
+fn calculate_reward<'address>(
     total_stake: u64,
     stake_per_voter: &HashMap<&'address Address, u64>,
-) -> HashMap<&'address Address, U64F64> {
+    active_addresses: &ActiveAddresses,
+    total_rewards: Rewards,
+) -> HashMap<&'address Address, Rewards> {
     stake_per_voter
         .iter()
-        .map(|(k, v)| (*k, fixed::types::U64F64::from_num(*v) / total_stake as u128))
+        .map(|(k, v)| {
+            // if it doesnt appear in the votes count, it means it did not vote
+            let reward = if active_addresses.contains(k) {
+                Rewards::from(*v) / Rewards::from(total_stake) * total_rewards
+            } else {
+                Rewards::ZERO
+            };
+            (*k, reward)
+        })
         .collect()
 }
 
-/// get the proportional reward from a share and total rewards amount
-pub fn reward_from_share(share: U64F64, total_reward: u64) -> fixed::types::U64F64 {
-    fixed::types::U64F64::from_num(total_reward) * share
+pub type VoteCount = HashMap<String, u64>;
+pub type ActiveAddresses = HashSet<Address>;
+
+fn active_addresses(
+    vote_count: VoteCount,
+    block0: &Block0Configuration,
+    threshold: u64,
+) -> ActiveAddresses {
+    let discrimination = block0.blockchain_configuration.discrimination;
+    vote_count
+        .into_iter()
+        .filter_map(|(account_hex, count)| {
+            if count >= threshold {
+                Some(
+                    account_hex_to_address(account_hex, discrimination)
+                        .expect("Valid hex encoded UnspecifiedAccountIdentifier"),
```

Comment:
Is that OK to panic there? I mean, a failure with one address would mean that we got a corrupted file anyway, but I would love to have better error messaging in the future.